### PR TITLE
lomiri.lomiri-ui-extras: init at 0.6.2

### PR DESCRIPTION
**Pull request by OPNA2608**
## Description of changes

Working towards #99090.

[Lomiri UI Extras](https://gitlab.com/ubports/development/core/lomiri-ui-extras), more QML components for Lomiri applications. Required by some Lomiri core applications - gallery, file manager, terminal, web browser.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by OPNA2608** [comment]
Seems like the exiv2 bump in #231244 breaks this on master:

```
[  1%] Building CXX object modules/Lomiri/Components/Extras/plugin/CMakeFiles/lomiri-ui-extras-plugin.dir/lomiri-ui-extras-plugin_autogen/mocs_compilation.cpp.o
In file included from /build/source/build/modules/Lomiri/Components/Extras/plugin/lomiri-ui-extras-plugin_autogen/OHEM2FCV2D/moc_photo-metadata.cpp:10,
                 from /build/source/build/modules/Lomiri/Components/Extras/plugin/lomiri-ui-extras-plugin_autogen/mocs_compilation.cpp:7:
/build/source/build/modules/Lomiri/Components/Extras/plugin/lomiri-ui-extras-plugin_autogen/OHEM2FCV2D/../../../../../../../../modules/Lomiri/Components/Extras/plugin/photoeditor/photo-metadata.h:62:19: error: 'AutoPtr' in 'class Exiv2::Image' does not name a type
   62 |     Exiv2::Image::AutoPtr m_image;
      |                   ^~~~~~~
make[2]: *** [modules/Lomiri/Components/Extras/plugin/CMakeFiles/lomiri-ui-extras-plugin.dir/build.make:76: modules/Lomiri/Components/Extras/plugin/CMakeFiles/lomiri-ui-extras-plugin.dir/lomiri-ui-extras-plugin_autogen/mocs_compilation.cpp.o] Error 1
```

**Comment by OPNA2608** [comment]
`tst_PhotoEditorPhoto` appears to be flaky, need to look into it (`lomiri-terminal-app` PR keeps FTB cus of it).

File: pkgs/desktops/lomiri/qml/lomiri-ui-extras/default.nix [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+  ];
+
+  checkInputs = [
+    lomiri-ui-toolkit
+  ];
+
+  dontWrapQtApps = true;
+
+  cmakeFlags = [
+    "-DENABLE_TESTS=${lib.boolToString finalAttrs.doCheck}"
```
Comment (SuperSandro2000) [inline pkgs/desktops/lomiri/qml/lomiri-ui-extras/default.nix:92]:
```suggestion
    "-DENABLE_TESTS=${lib.boolToString finalAttrs.finalPackage.doCheck}"
```

**Review by SuperSandro2000 — APPROVED** [review]

File: pkgs/desktops/lomiri/qml/lomiri-ui-extras/default.nix (hunk 1 above)
Comment (OPNA2608) [reply pkgs/desktops/lomiri/qml/lomiri-ui-extras/default.nix]: Applied.

**Review by SuperSandro2000 — APPROVED** [review]
not sure if we should squash the commits or not, up to you. LGTM, please merge

**Comment by OPNA2608** [comment]
> not sure if we should squash the commits or not, up to you.

Squashed.

> LGTM, please merge

I don't have merge permissions.

**Review by SuperSandro2000 — APPROVED** [review]

**Review by pbsds — APPROVED** [review]
Result of `nixpkgs-review pr 247101` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>lomiri.lomiri-ui-extras</li>
  </ul>
</details>


- [x] package path fits guidelines
- [x] package name fits guidelines
- [x] package version fits guidelines
- [x] package builds on x86_64-linux
- [ ] executables tested on x86_64-linux
  - just a lib, but `env -i ldd` is happy
- [x] `meta.description` is set and fits guidelines
- [x] `meta.license` fits upstream license
- [x] `meta.platforms` is set
- [x] `meta.maintainers` is set
- [x] build time only dependencies are declared in `nativeBuildInputs`
- [x] source is fetched using the appropriate function
- [x] the list of `phases` is not overridden
- [x] when a phase (like `installPhase`) is overridden it starts with `runHook preInstall` and ends with `runHook postInstall`.
- [x] patches have a comment describing either the upstream URL or a reason why the patch wasn't upstreamed
- [x] patches that are remotely available are fetched rather than vendored

LGTM

**Pull request merged**